tools/ci-trigger/pr.go

```diff
@@ -151,6 +158,41 @@ func (p *pullRequest) createBuild(ctx context.Context, buildClient *cloudbuild.S
 				}
 			}
 		}
+	physicalDeviceLoop:
+		for i, physicalDevice := range p.Physical {
+			if physicalDevice.Type == d {
+				if len(physicalDevice.Tests) == 0 {
+					continue
+				}
+				for _, v := range physicalDevice.Tests {
+					if v.Status != "pending authorization" {
+						continue physicalDeviceLoop
+					}
+				}
+				jobID, err := uuid.NewRandom()
+				if err != nil {
+					return fmt.Errorf("uuid.NewRandom device %q: %w", physicalDevice.Type.String(), err)
+				}
+				p.Physical[i].ArchivePath = objPath
+				p.Physical[i].CloudBuildID = jobID.String()
+				vendor := strings.ToLower(physicalDevice.Type.Vendor.String())
+				vendor = strings.ReplaceAll(vendor, " ", "")
+				p.Physical[i].CloudBuildRawLogURL = fmt.Sprintf("https://storage.cloud.google.com/featureprofiles-ci-logs-%s/log-%s.txt", vendor, jobID)
+				jsonMsg, err := json.Marshal(p.Physical[i])
+				if err != nil {
+					return fmt.Errorf("json.Marshal device %q: %w", physicalDevice.Type.String(), err)
+				}
+				result := pubsubTopic.Publish(ctx, &pubsub.Message{Data: jsonMsg})
+				id, err := result.Get(ctx)
+				if err != nil {
+					return fmt.Errorf("pubsubTopic.Publish device %q: %w", physicalDevice.Type.String(), err)
+				}
+				glog.Infof("Sent Physical Test Job for PR%d at commit %q for device %q via %s", p.ID, p.HeadSHA, physicalDevice.Type.String(), id)
+				for j := range physicalDevice.Tests {
```

Comment:
iterator over the values instead, i.e.

for _, t := range physicalDevice.Tests {
  t.Status = "setup"
}

Reply:
Done